diff --git a/wp-includes/media.php b/wp-includes/media.php
index 743e8d7..ddcaf69 100644
--- a/wp-includes/media.php
+++ b/wp-includes/media.php
@@ -3062,9 +3062,6 @@ function wp_image_editor_supports( $args = array() ) {
  *                     editor claims to support the request.
  */
 function _wp_image_editor_choose( $args = array() ) {
-	require_once ABSPATH . WPINC . '/class-wp-image-editor.php';
-	require_once ABSPATH . WPINC . '/class-wp-image-editor-gd.php';
-	require_once ABSPATH . WPINC . '/class-wp-image-editor-imagick.php';
 	/**
 	 * Filters the list of image editing library classes.
 	 *
diff --git a/wp-includes/wp-diff.php b/wp-includes/wp-diff.php
index deae250..a3e68bf 100644
--- a/wp-includes/wp-diff.php
+++ b/wp-includes/wp-diff.php
@@ -17,5 +17,3 @@ if ( ! class_exists( 'Text_Diff', false ) ) {
 	require( ABSPATH . WPINC . '/Text/Diff/Renderer/inline.php' );
 }
 
-require( ABSPATH . WPINC . '/class-wp-text-diff-renderer-table.php' );
-require( ABSPATH . WPINC . '/class-wp-text-diff-renderer-inline.php' );
diff --git a/wp-settings.php b/wp-settings.php
index 518db55..95fc316 100644
--- a/wp-settings.php
+++ b/wp-settings.php
@@ -15,15 +15,42 @@
  */
 define( 'WPINC', 'wp-includes' );
 
+spl_autoload_register(
+    static function ($class) {
+        $basename = strtolower(str_replace('_', '-', $class)) . '.php';
+
+        // Possible locations in separate if to spare costs.
+        $path = ABSPATH . WPINC . '/class-' . $basename;
+        if (file_exists($path)) {
+            require_once $path;
+            return;
+        }
+
+        $path = ABSPATH . WPINC . '/class-wp-' . $basename;
+        if (file_exists($path)) {
+            require_once $path;
+            return;
+        }
+
+        $path = ABSPATH . WPINC . '/class-' . str_replace('wp-', '', $basename);
+        if (file_exists($path)) {
+            require_once $path;
+            return;
+        }
+
+        $path = ABSPATH . WPINC . '/' . str_replace('_', '/', $class). '.php';
+        if (file_exists($path)) {
+            require_once $path;
+        }
+    }
+);
+
+register_shutdown_function(function () {
+    var_dump(error_get_last());
+});
+
 // Include files required for initialization.
 require( ABSPATH . WPINC . '/load.php' );
-require( ABSPATH . WPINC . '/class-wp-paused-extensions-storage.php' );
-require( ABSPATH . WPINC . '/class-wp-fatal-error-handler.php' );
-require( ABSPATH . WPINC . '/class-wp-recovery-mode-cookie-service.php' );
-require( ABSPATH . WPINC . '/class-wp-recovery-mode-key-service.php' );
-require( ABSPATH . WPINC . '/class-wp-recovery-mode-link-service.php' );
-require( ABSPATH . WPINC . '/class-wp-recovery-mode-email-service.php' );
-require( ABSPATH . WPINC . '/class-wp-recovery-mode.php' );
 require( ABSPATH . WPINC . '/error-protection.php' );
 require( ABSPATH . WPINC . '/default-constants.php' );
 require_once( ABSPATH . WPINC . '/plugin.php' );
@@ -105,14 +132,9 @@ wp_set_lang_dir();
 
 // Load early WordPress files.
 require( ABSPATH . WPINC . '/compat.php' );
-require( ABSPATH . WPINC . '/class-wp-list-util.php' );
 require( ABSPATH . WPINC . '/formatting.php' );
 require( ABSPATH . WPINC . '/meta.php' );
 require( ABSPATH . WPINC . '/functions.php' );
-require( ABSPATH . WPINC . '/class-wp-meta-query.php' );
-require( ABSPATH . WPINC . '/class-wp-matchesmapregex.php' );
-require( ABSPATH . WPINC . '/class-wp.php' );
-require( ABSPATH . WPINC . '/class-wp-error.php' );
 require( ABSPATH . WPINC . '/pomo/mo.php' );
 
 // Include the wpdb class and, if present, a db.php database drop-in.
@@ -131,8 +153,6 @@ require( ABSPATH . WPINC . '/default-filters.php' );
 
 // Initialize multisite if enabled.
 if ( is_multisite() ) {
-	require( ABSPATH . WPINC . '/class-wp-site-query.php' );
-	require( ABSPATH . WPINC . '/class-wp-network-query.php' );
 	require( ABSPATH . WPINC . '/ms-blogs.php' );
 	require( ABSPATH . WPINC . '/ms-settings.php' );
 } elseif ( ! defined( 'MULTISITE' ) ) {
@@ -148,53 +168,30 @@ if ( SHORTINIT ) {
 
 // Load the L10n library.
 require_once( ABSPATH . WPINC . '/l10n.php' );
-require_once( ABSPATH . WPINC . '/class-wp-locale.php' );
-require_once( ABSPATH . WPINC . '/class-wp-locale-switcher.php' );
 
 // Run the installer if WordPress is not installed.
 wp_not_installed();
 
 // Load most of WordPress.
-require( ABSPATH . WPINC . '/class-wp-walker.php' );
-require( ABSPATH . WPINC . '/class-wp-ajax-response.php' );
 require( ABSPATH . WPINC . '/capabilities.php' );
-require( ABSPATH . WPINC . '/class-wp-roles.php' );
-require( ABSPATH . WPINC . '/class-wp-role.php' );
-require( ABSPATH . WPINC . '/class-wp-user.php' );
-require( ABSPATH . WPINC . '/class-wp-query.php' );
 require( ABSPATH . WPINC . '/query.php' );
 require( ABSPATH . WPINC . '/date.php' );
 require( ABSPATH . WPINC . '/theme.php' );
-require( ABSPATH . WPINC . '/class-wp-theme.php' );
 require( ABSPATH . WPINC . '/template.php' );
 require( ABSPATH . WPINC . '/user.php' );
-require( ABSPATH . WPINC . '/class-wp-user-query.php' );
-require( ABSPATH . WPINC . '/class-wp-session-tokens.php' );
-require( ABSPATH . WPINC . '/class-wp-user-meta-session-tokens.php' );
-require( ABSPATH . WPINC . '/class-wp-metadata-lazyloader.php' );
 require( ABSPATH . WPINC . '/general-template.php' );
 require( ABSPATH . WPINC . '/link-template.php' );
 require( ABSPATH . WPINC . '/author-template.php' );
 require( ABSPATH . WPINC . '/post.php' );
-require( ABSPATH . WPINC . '/class-walker-page.php' );
-require( ABSPATH . WPINC . '/class-walker-page-dropdown.php' );
-require( ABSPATH . WPINC . '/class-wp-post-type.php' );
-require( ABSPATH . WPINC . '/class-wp-post.php' );
 require( ABSPATH . WPINC . '/post-template.php' );
 require( ABSPATH . WPINC . '/revision.php' );
 require( ABSPATH . WPINC . '/post-formats.php' );
 require( ABSPATH . WPINC . '/post-thumbnail-template.php' );
 require( ABSPATH . WPINC . '/category.php' );
-require( ABSPATH . WPINC . '/class-walker-category.php' );
-require( ABSPATH . WPINC . '/class-walker-category-dropdown.php' );
 require( ABSPATH . WPINC . '/category-template.php' );
 require( ABSPATH . WPINC . '/comment.php' );
-require( ABSPATH . WPINC . '/class-wp-comment.php' );
-require( ABSPATH . WPINC . '/class-wp-comment-query.php' );
-require( ABSPATH . WPINC . '/class-walker-comment.php' );
 require( ABSPATH . WPINC . '/comment-template.php' );
 require( ABSPATH . WPINC . '/rewrite.php' );
-require( ABSPATH . WPINC . '/class-wp-rewrite.php' );
 require( ABSPATH . WPINC . '/feed.php' );
 require( ABSPATH . WPINC . '/bookmark.php' );
 require( ABSPATH . WPINC . '/bookmark-template.php' );
@@ -203,31 +200,13 @@ require( ABSPATH . WPINC . '/cron.php' );
 require( ABSPATH . WPINC . '/deprecated.php' );
 require( ABSPATH . WPINC . '/script-loader.php' );
 require( ABSPATH . WPINC . '/taxonomy.php' );
-require( ABSPATH . WPINC . '/class-wp-taxonomy.php' );
-require( ABSPATH . WPINC . '/class-wp-term.php' );
-require( ABSPATH . WPINC . '/class-wp-term-query.php' );
-require( ABSPATH . WPINC . '/class-wp-tax-query.php' );
 require( ABSPATH . WPINC . '/update.php' );
 require( ABSPATH . WPINC . '/canonical.php' );
 require( ABSPATH . WPINC . '/shortcodes.php' );
 require( ABSPATH . WPINC . '/embed.php' );
-require( ABSPATH . WPINC . '/class-wp-embed.php' );
-require( ABSPATH . WPINC . '/class-oembed.php' );
-require( ABSPATH . WPINC . '/class-wp-oembed-controller.php' );
 require( ABSPATH . WPINC . '/media.php' );
 require( ABSPATH . WPINC . '/http.php' );
-require( ABSPATH . WPINC . '/class-http.php' );
-require( ABSPATH . WPINC . '/class-wp-http-streams.php' );
-require( ABSPATH . WPINC . '/class-wp-http-curl.php' );
-require( ABSPATH . WPINC . '/class-wp-http-proxy.php' );
-require( ABSPATH . WPINC . '/class-wp-http-cookie.php' );
-require( ABSPATH . WPINC . '/class-wp-http-encoding.php' );
-require( ABSPATH . WPINC . '/class-wp-http-response.php' );
-require( ABSPATH . WPINC . '/class-wp-http-requests-response.php' );
-require( ABSPATH . WPINC . '/class-wp-http-requests-hooks.php' );
 require( ABSPATH . WPINC . '/widgets.php' );
-require( ABSPATH . WPINC . '/class-wp-widget.php' );
-require( ABSPATH . WPINC . '/class-wp-widget-factory.php' );
 require( ABSPATH . WPINC . '/nav-menu.php' );
 require( ABSPATH . WPINC . '/nav-menu-template.php' );
 require( ABSPATH . WPINC . '/admin-bar.php' );
@@ -258,9 +237,6 @@ require( ABSPATH . WPINC . '/rest-api/fields/class-wp-rest-term-meta-fields.php'
 require( ABSPATH . WPINC . '/rest-api/fields/class-wp-rest-user-meta-fields.php' );
 require( ABSPATH . WPINC . '/rest-api/search/class-wp-rest-search-handler.php' );
 require( ABSPATH . WPINC . '/rest-api/search/class-wp-rest-post-search-handler.php' );
-require( ABSPATH . WPINC . '/class-wp-block-type.php' );
-require( ABSPATH . WPINC . '/class-wp-block-type-registry.php' );
-require( ABSPATH . WPINC . '/class-wp-block-parser.php' );
 require( ABSPATH . WPINC . '/blocks.php' );
 require( ABSPATH . WPINC . '/blocks/archives.php' );
 require( ABSPATH . WPINC . '/blocks/block.php' );
